src/append/rolling_file/rotation.rs

```diff
@@ -31,71 +32,79 @@ pub enum Rotation {
-            Rotation::Hourly => *current_date + Duration::hours(1),
-            Rotation::Daily => *current_date + Duration::days(1),
             Rotation::Never => return None,
-        };
-
-        Some(self.round_date(&next_date).unix_timestamp() as usize)
-    }
-
-    fn round_date(&self, date: &OffsetDateTime) -> OffsetDateTime {
-        match *self {
             Rotation::Minutely => {
-                let time = Time::from_hms(date.hour(), date.minute(), 0)
-                    .expect("invalid time; this is a bug in logforth rolling file appender");
-                date.replace_time(time)
+                (*current_date + 1.minute()).round(timestamp_round.smallest(Unit::Minute))
             }
             Rotation::Hourly => {
-                let time = Time::from_hms(date.hour(), 0, 0)
-                    .expect("invalid time; this is a bug in logforth rolling file appender");
-                date.replace_time(time)
-            }
-            Rotation::Daily => {
-                let time = Time::from_hms(0, 0, 0)
-                    .expect("invalid time; this is a bug in logforth rolling file appender");
-                date.replace_time(time)
+                (*current_date + 1.hour()).round(timestamp_round.smallest(Unit::Hour))
             }
-            Rotation::Never => unreachable!("Rotation::Never is impossible to round."),
+            Rotation::Daily => (*current_date + 1.day()).round(timestamp_round.smallest(Unit::Day)),
         }
+        .expect("invalid time; this is a bug in logforth rolling file appender");
+
+        println!("next_date: {:?}", next_date);
+
+        Some(next_date.as_millisecond() as usize)
     }
 
-    pub fn date_format(&self) -> Vec<format_description::FormatItem<'static>> {
+    pub fn date_format(&self) -> &'static str {
         match *self {
-            Rotation::Minutely => format_description::parse("[year]-[month]-[day]-[hour]-[minute]"),
-            Rotation::Hourly => format_description::parse("[year]-[month]-[day]-[hour]"),
-            Rotation::Daily => format_description::parse("[year]-[month]-[day]"),
-            Rotation::Never => format_description::parse("[year]-[month]-[day]"),
+            Rotation::Minutely => "%F-%H-%M",
+            Rotation::Hourly => "%F-%H",
+            Rotation::Daily => "%F",
+            Rotation::Never => "%F",
         }
-        .expect("failed to create a formatter; this is a bug in logforth rolling file appender")
     }
 }
 
 #[cfg(test)]
 mod tests {
-    use time::macros::datetime;
+    use jiff::civil::date;
+    use jiff::tz::Offset;
+    use jiff::tz::TimeZone;
 
     use super::Rotation;
 
     #[test]
     fn test_next_date_timestamp() {
-        let current_date = datetime!(2024-08-10 17:12:52 +8);
+        let current_date = date(2024, 8, 10)
+            .at(17, 12, 52, 0)
+            .to_zoned(TimeZone::fixed(Offset::constant(8)))
+            .unwrap();
```

Comment:
If we can have some convenient constructor to make these dates?

Reply:
You can shorten it slightly here with `jiff::tz::offset(8).to_time_zone()`, but otherwise this is intended to be the "convenient" way of building these datetimes. You could also parse it, e.g., `"2024-08-10T17:12:52+08".parse::<Timestamp>().unwrap()`.

Reply:
@BurntSushi "2024-08-10T17:12:52+08"  failed to parse as `Zoned`:

```
let now = Zoned::from_str("2024-08-10T17:12:52+08").unwrap();
```

Does Zoned use other flavor? To overcome the add day issue I tend to use Zoned everywhere now.

```
called `Result::unwrap()` on an `Err` value: Error { inner: ErrorInner { kind: Adhoc(failed to find time zone in square brackets in "2024-08-10T17:12:52+08", which is required for parsing a zoned instant), cause: None } }
```

Reply:
Use `Zoned::new("2024-08-10T17:12:52+08".parse().unwrap(), TimeZone::UTC);` to work around. Not sure if it's the best practice.

Reply:
Does the [documentation](https://docs.rs/jiff/latest/jiff/fmt/temporal/struct.DateTimeParser.html#example-parsing-without-an-iana-time-zone) answer your question here?

I would either add the `[+08]` annotation or parse into a timestamp and convert to a `Zoned` via `timestamp.to_zoned(jiff::tz::TimeZone::UTC)`.

Reply:
Thank you! It's very clear. I use the extra `[+08]` option.